avro-serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/deserializers/AWSDeserializerDataParser.java

```diff
@@ -64,19 +64,20 @@ public static AWSDeserializerDataParser getInstance() {
      *                                      schema registry
      */
     public UUID getSchemaVersionId(ByteBuffer byteBuffer) {
+        byteBuffer.rewind();
```

Comment:
Since you now slice the input buffer, why rewind?  I reckon you should leave the buffer without any side effects.

Actually, I’m not even sure this is the right approach.  I would argue that it should be the responsibility of the deserializer’s client, to handle buffer positions.  This is the KCL.  It ought to slice the buffer and use the slice to test for schema encoding.  If this test is positive then it should decode the schema, using another slice.  It is somewhat unfortunate that the deserializer interface has access to the “plain” data through a third method but so then the KCL ought to call that method, with a third slice.

So I don’t think that the deserializer (or this class) should be responsible for maintaining buffer positions…

Reply:
Thanks for reviewing!

>Since you now slice the input buffer, why rewind? I reckon you should leave the buffer without any side effects.

When a buffer with non-initial position is sliced, the sliced buffer starts from non-initial position and previous bytes are not accessible. Clients who might used to sending buffer with non-rewinded state assuming the GSR library will rewind might break, so kept it same. 

KCL is one of the consumers of the library, customers can directly use this library. We want this behavior consistent across all the clients so that every doesn't have to make this change. It's also good to hide the complexity from clients. Do you see any side-effects with this approach?

Reply:
I understand your concern regarding existing clients that might break if the code doesn't rewind the buffer... I hadn't considered this.